### PR TITLE
Add Undo/Redo support in the viewer

### PR DESCRIPTION
**Pull request by emilk**
### What
* Closes https://github.com/rerun-io/rerun/issues/3135
* Proceeded by https://github.com/rerun-io/rerun/pull/7602
* Proceeded by https://github.com/rerun-io/rerun/pull/7603
* Proceeded by https://github.com/rerun-io/rerun/pull/8241
* New issue: https://github.com/rerun-io/rerun/issues/8249

This PR implements Undo and Redo for any edit to the active blueprint.

https://github.com/user-attachments/assets/05018729-f01e-42f4-a84f-b48dbf31b060

### Implementation
This implements undo/redo by editing the "time cursor" for the blueprint timeline. Undo moves it backwards, redo forwards. When doing an action, all redo history is erased from the store with a new `ChunkStore::drop_time_range` function.

### Known shortcomings
* Undo doesn't work when the blueprint streams panel is open

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7546?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7546?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7546)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Comment by emilk** [comment]
This is blocked by:
* https://github.com/rerun-io/rerun/issues/8056

File: crates/viewer/re_viewport_blueprint/src/container.rs [hunk 1]
```diff
@@ -356,12 +374,25 @@ impl ContainerBlueprint {
...
+        let timepoint = ctx.store_context.blueprint_timepoint_for_writes();
+
+        let chunk = Chunk::builder(self.entity_path())
+            .with_archetype(
+                RowId::new(),
+                timepoint.clone(),
+                &re_types::archetypes::Clear::recursive(),
+            )
+            .build()
+            .expect("Failed to serialize Clear component !?");
```
Comment (teh-cmc) [inline crates/viewer/re_viewport_blueprint/src/container.rs:389]: `.expect` is still unnecessary risk here -- why not check the error?

File: crates/viewer/re_viewport_blueprint/src/space_view.rs [hunk 2]
```diff
@@ -310,10 +310,24 @@ impl SpaceViewBlueprint {
...
+        let timepoint = ctx.store_context.blueprint_timepoint_for_writes();
+
+        let chunk = Chunk::builder(self.entity_path())
+            .with_archetype(
+                RowId::new(),
+                timepoint.clone(),
+                &re_types::archetypes::Clear::recursive(),
+            )
+            .build()
+            .expect("Failed to serialize Clear component !?");
```
Comment (teh-cmc) [inline crates/viewer/re_viewport_blueprint/src/space_view.rs:324]: Ditto (a helper would be nice indeed).

File: crates/viewer/re_viewer/src/app_state.rs [hunk 3]
```diff
@@ -31,9 +33,12 @@ pub struct AppState {
     recording_configs: HashMap<StoreId, RecordingConfig>,
     pub blueprint_cfg: RecordingConfig,
 
+    /// Maps blueperint id to the current undo state for it.
```
Comment (teh-cmc) [inline crates/viewer/re_viewer/src/app_state.rs:36]:
```suggestion
    /// Maps blueprint id to the current undo state for it.

**Review by teh-cmc — APPROVED** [review]

**Pull request merged**